src/consumers.ts

```diff
@@ -31,6 +31,8 @@ interface IConsumer
 {
     binding: IBindingDescriptor;
     item: any;
+    vulnerabilityCount: any;
+    advisoryCount: any;
```

Comment:
These are number type, not any

Reply:
BTW, why do you need to put inside the baseclass? Other properties are not added to it right?

Reply:
I can change to the number type.

Reply:
> BTW, why do you need to put inside the baseclass? Other properties are not added to it right?

According to me; the `items` in `DiagnosticPipeline` is a type of  `Array<IPipelineItem<Diagnostic[]>>`
```
class DiagnosticsPipeline implements IPipeline<Diagnostic[]>
{
    items: Array<IPipelineItem<Diagnostic[]>>;
    ...
}
```
and `IPipelineItem` extends `IConsumer` and `IProducer<T>`
`interface IPipelineItem<T> extends IConsumer, IProducer<T> {};`

And if we want to access any properties of `items` via reference of `DiagnosticsPipeline`, so that property should be there in `IConsumer` or `IPipelineItem<T>` or `IProducer<T>`.

Correct other properties are not there, if we want to add other properties in `IConsumer`, we can add those.

Reply:
Please correct me if i am wrong.
if you have other better solution please let me know.

Reply:
> I can change to the number type.

PTAL : https://github.com/fabric8-analytics/fabric8-analytics-lsp-server/commit/0391ba4a038ba374107d78e1952d7d2b92367ab8

Reply:
You can typecast the `items` to `SecurityEngine` and avoid declaring these in the base class.